packages/jsii-rosetta/test/commands/extract.test.ts

```diff
@@ -59,6 +59,45 @@ beforeEach(() => {
 
 afterEach(() => assembly.cleanup());
 
+test('extract works with zipped assembly files', async () => {
+  const assembly = TestJsiiModule.fromSource(
+    {
+      'index.ts': `
+      export class ClassA {
+        public someMethod() {
+        }
+      }
+      export class ClassB {
+        public anotherMethod() {
+        }
+      }
+      `,
+      'README.md': DUMMY_README,
+    },
+    {
+      name: 'my_assembly',
+      jsii: DUMMY_JSII_CONFIG,
+    },
+    true /* zipAssemblyFile */,
```

Comment:
Pass a property bag instead so you don't have to comment everywhere... This signature is starting to look a little awkward, too... maybe `TestJsiiModule` could have a static `.enableAssemblyCompression()` that returns a version of itself that enables compression.